src/services/RewardService.ts

```diff
@@ -0,0 +1,182 @@
+import axios from 'axios';
+import { RewardOption, RewardTypes } from 'src/types';
+import offerRepository from '../repositories/OfferRepository';
+import ApiError from '../exceptions/ApiError';
+import dealRepository from '../repositories/DealRepository';
+
+const options = {
+  LIF: {
+    id: 'winding-tree',
+    symbol: 'lif',
+    name: 'Lif'
+  },
+  NCT: {
+    id: 'toucan-protocol-nature-carbon-tonne',
+    symbol: 'nct',
+    name: 'Toucan Protocol: Nature Carbon Tonne'
+  }
+};
+
+const rewardPercentage = 6;
+
+// Note: Free API is limited to 50 calls/minute.
+const apiURL = 'https://api.coingecko.com/api/v3/';
+const coinGeckoQuoteCurrencies = [
+  'btc',
+  'eth',
+  'ltc',
+  'bits',
+  'bch',
+  'bnb',
+  'eos',
+  'xrp',
+  'xlm',
+  'link',
+  'dot',
+  'yfi',
+  'usd',
+  'aed',
+  'ars',
+  'aud',
+  'bdt',
+  'bhd',
+  'bmd',
+  'brl',
+  'cad',
+  'chf',
+  'clp',
+  'cny',
+  'czk',
+  'dkk',
+  'eur',
+  'gbp',
+  'hkd',
+  'huf',
+  'idr',
+  'ils',
+  'inr',
+  'jpy',
+  'krw',
+  'kwd',
+  'lkr',
+  'mmk',
+  'mxn',
+  'myr',
+  'ngn',
+  'nok',
+  'nzd',
+  'php',
+  'pkr',
+  'pln',
+  'rub',
+  'sar',
+  'sek',
+  'sgd',
+  'thb',
+  'try',
+  'twd',
+  'uah',
+  'vef',
+  'vnd',
+  'zar',
+  'xdr',
+  'xag',
+  'xau',
+  'sats'
+];
+
+export class RewardService {
+  public async getOptions(offerId: string): Promise<RewardOption[]> {
+    const offer = await offerRepository.getOne(offerId);
```

Comment:
don't use offers here, offers is temp data, lifetime of offer 30min, use deal, deal includes offer